src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -114,6 +121,35 @@ public function supportsClass($class)
         return 'Symfony\Component\Security\Core\User\User' === $class;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    {
+        if (!$user instanceof User) {
+            throw new UnsupportedUserException(sprintf('Instances of "%s" are not supported.', \get_class($user)));
+        }
+
+        if (null === $this->passwordAttribute) {
+            return;
+        }
+
+        try {
+            // Tell loadUserByUsername() to keep the $entry around
+            $this->entry = true;
+
+            if ($user->isEqualTo($this->loadUserByUsername($user->getUsername())) && \is_object($this->entry)) {
```

Comment:
How could `isEqualTo()` be `false` here?

Reply:
> How could isEqualTo() be false here?

when a concurrent request updates the user

> Won’t $this->entry always be true thanks to the line above

no because `loadUserByUsername()` changes its state

> not sure I love using one property to store a Boolean flag and/or an object

I don't *love* it either, but that's still needed, and internal.

Reply:
We could completely remove the need for reloading the user and get rid of the `$entry` property if we had a `LdapUser` aware of its ldap `entry`: #32824

Reply:
Cool, don't forget to update your PR once this one is merged :)

Reply:
`$this->entry now gone`